Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -130,6 +137,99 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
         AppEnvironment.current.environmentType.stripePublishableKey
       )
     }
+
+    // MARK: Validate Checkout Details On Submit
+
+    // MARK: - Validate Existing Cards
+
+    /// Capture current users stored credit cards in the case that we need to validate an existing payment method
+    let storedCardsEvent = initialData.ignoreValues()
+      .switchMap { _ in
+        AppEnvironment.current.apiService
+          .fetchGraphUser(withStoredCards: true)
+          .ksr_debounce(.seconds(1), on: AppEnvironment.current.scheduler)
+          .map { envelope in (envelope, false) }
+          .prefix(value: (nil, true))
+          .materialize()
+      }
+
+    let storedCardsValues = storedCardsEvent.values()
+      .filter(second >>> isFalse)
+      .map(first)
+      .skipNil()
+      .map { $0.me.storedCards.storedCards }
+
+    let selectedExistingCard = self.creditCardSelectedProperty.signal.skipNil()
+      .filter { $0.isNewPaymentMethod == false }
+
+    let newPaymentIntentEvent = initialData
```

Comment:
This only needs to happen when you're _not_ adding a new card via the payment method selector, since in that case, we've already made a PI for you. Should this be attached to a different event than `initialData`?